src/Bicep.Core/Configuration/RootConfiguration.cs

```diff
@@ -106,5 +107,33 @@ public string ToUtf8Json()
 
             return Encoding.UTF8.GetString(bufferWriter.WrittenSpan);
         }
+
+        private static string? ExpandCacheRootDirectory(string? cacheRootDirectory)
+        {
+            /*
+             Note: the method is a simple workaround for https://github.com/Azure/bicep/issues/10935. To reduce
+             complexity, it does not handle all cross-platform edge cases, such as "~username" on Unix based systems.
+             In the future, we may want to read CacheRootDirectory from a environment variable instead whose value
+             must be a full path.
+            */
+            if (string.IsNullOrEmpty(cacheRootDirectory) || cacheRootDirectory[0] != '~')
```

Comment:
We don't need to support `%USERPROFILE%` or anything like that to consider this cross-platform, do we?

Reply:
No, we don't.